src/psyclone/tests/dynamo0p3_builtins_test.py

```diff
@@ -495,73 +478,66 @@ def test_aX_plus_Y(monkeypatch, annexed):
-        code = str(psy.gen)
-        print(code)
-        if not distmem:
-            output = (
-                "    SUBROUTINE invoke_0(f3, a, f1, f2)\n"
-                "      REAL(KIND=r_def), intent(in) :: a\n"
-                "      TYPE(field_type), intent(inout) :: f3\n"
-                "      TYPE(field_type), intent(in) :: f1, f2\n"
-                "      INTEGER df\n"
-                "      INTEGER ndf_any_space_1_f3, undf_any_space_1_f3\n"
-                "      INTEGER nlayers\n"
-                "      TYPE(field_proxy_type) f3_proxy, f1_proxy, f2_proxy\n"
-                "      !\n"
-                "      ! Initialise field and/or operator proxies\n"
-                "      !\n"
-                "      f3_proxy = f3%get_proxy()\n"
-                "      f1_proxy = f1%get_proxy()\n"
-                "      f2_proxy = f2%get_proxy()\n"
-                "      !\n"
-                "      ! Initialise number of layers\n"
-                "      !\n"
-                "      nlayers = f3_proxy%vspace%get_nlayers()\n"
-                "      !\n"
-                "      ! Initialise number of DoFs for any_space_1_f3\n"
-                "      !\n"
-                "      ndf_any_space_1_f3 = f3_proxy%vspace%get_ndf()\n"
-                "      undf_any_space_1_f3 = f3_proxy%vspace%get_undf()\n"
-                "      !\n"
-                "      ! Call our kernels\n"
-                "      !\n"
-                "      DO df=1,undf_any_space_1_f3\n"
-                "        f3_proxy%data(df) = a*f1_proxy%data(df) + "
-                "f2_proxy%data(df)\n"
-                "      END DO \n"
-                "      !\n"
-                "    END SUBROUTINE invoke_0\n")
-            assert output in code
-        else:
-            mesh_code_present("f3", code)
-            output_dm_2 = (
-                "      !\n"
-                "      ! Call kernels and communication routines\n"
-                "      !\n"
-                "      DO df=1,f3_proxy%vspace%get_last_dof_annexed()\n"
-                "        f3_proxy%data(df) = a*f1_proxy%data(df) + "
-                "f2_proxy%data(df)\n"
-                "      END DO \n"
-                "      !\n"
-                "      ! Set halos dirty/clean for fields modified in the "
-                "above loop\n"
-                "      !\n"
-                "      CALL f3_proxy%set_dirty()\n"
-                "      !\n")
-            if not annexed:
-                output_dm_2 = output_dm_2.replace("dof_annexed", "dof_owned")
-            print(output_dm_2)
-            assert output_dm_2 in code
-
-
-def test_inc_aX_plus_Y(monkeypatch, annexed):
+
+    psy = PSyFactory(API, distributed_memory=dist_mem).create(invoke_info)
+    # Test string method
+    first_invoke = psy.invokes.invoke_list[0]
+    kern = first_invoke.schedule.children[0].children[0]
+    assert str(kern) == "Built-in: aX_plus_Y"
+    # Test code generation
+    code = str(psy.gen)
+    print(code)
```

Comment:
There are print's in the updated tests but they were there before and we used to put them in so I'll ignore this.

Reply:
Yes, I noticed that but then I found I was making heavy use of them in this ticket so have left them alone.